src/app/loadout/mod-assignment-utils.ts

```diff
@@ -563,6 +599,53 @@ function calculateEnergyChange(
     : energyUsedAndWasted;
 }
 
+/**
+ * Calculates the total number of active conditional mods on the item.
+ * Used to ensure mod assignments favor results that activate these mods.
+ */
+function calculateTotalActivatedMods(
+  assignedModsForItem: PluggableInventoryItemDefinition[],
+  allAssignedMods: PluggableInventoryItemDefinition[]
+) {
+  let activeMods = 0;
+
+  for (const mod of assignedModsForItem) {
+    if (isPlugActive(mod, assignedModsForItem, allAssignedMods)) {
+      activeMods++;
+    }
+  }
+
+  return activeMods;
+}
+
+/**
+ * Determines whether a conditional mod has had its requirements met by the other mods.
+ * Currently only used for radiant light and powerful friends
+ */
+function isPlugActive(
```

Comment:
Didn't we have this somewhere else to determine whether the conditional stats were active on these?

Reply:
Yes, I basically took that function and pulled the parts I needed out of it. I nearly added this to be right along side it, but I only need it in this one place, and I don't currently see a use case for it outside of this mod algorithm.